api/artifact_lister.go

```diff
@@ -0,0 +1,137 @@
+package api
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+)
+
+type bitriseAPIClient interface {
```

Comment:
We generally have the following code structuring rules for steps:
```
// Every struct (that is logic) has a corresponding interface.
type ArtifactLister interface {
}

// Has the corresponding struct as unexported without `default` prefix
type artifactLister struct {
}

// And the constructor gives back the interface
func NewArtifactLister() ArtifactLister {
}

// This way we eliminate the possibility to create the struct without using the constructors. We are aware this is not the "golang standard" but decided to go in that direction for these reasons.
````

Reply:
I'm not saying it has to be refactored to this convention, maybe at some point in the future.